### PR TITLE
Topic outlining feature gaps in Serverless clusters.

### PR DESCRIPTION
**Pull request by ianjevans**
Fixes DOC-3650.

A new topic that explicitly outlines which features are either fully or partially supported in Serverless clusters.

### CODE REVIEW

**Comment by kannanlakshmi** [comment]
@ianjevans mind removing and adding Vivek B as the reviewer as the PM owner on Serverless? Thanks!

**Review by awoods187 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14282)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abhinavg6, @amruss, @emily-horing, @ianjevans, @mwang1026, and @thtruo)

___
*[`cockroachcloud/serverless-unsupported-features.md` line 8 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4bxfH88J3kG3Yjklp-:-N4bxfH88J3kG3Yjklp0:b-xkocdw) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L8)):*
> ```Markdown
> ---
> 
> {{ site.data.products.serverless }} is a [managed multi-tenant deployment of CockroachDB](architecture.html#cockroachdb-serverless) that automatically scales up and down based on the load on the cluster. {{ site.data.products.serverless }} works with almost all workloads that CockroachDB supports, but there are feature differences between {{ site.data.products.core }} or {{ site.data.products.dedicated }} clusters and {{ site.data.products.serverless }} clusters. This topic describes the features that are either unsupported or partially supported in {{ site.data.products.serverless }} clusters. Cockroach Labs intends to eliminate these feature gaps in future releases of {{ site.data.products.serverless }}.
> ```

Philosophical question. Do we want to use feature differences here? I'm thinking maybe something like "user experience differences" because we deliver this same UX via different features. I don't think its important to mention a feature not in serverless, but in crdb, when the UX the feature provides is provided by another feature, if it is, we should at least separate that out from other unsupported UX differences. 
___
*[`cockroachcloud/serverless-unsupported-features.md` line 26 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4by5RbAY64zacQBva7:-N4by5RbAY64zacQBva8:backbzl) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L26)):*
> ```Markdown
> Both {{ site.data.products.serverless }} and {{ site.data.products.dedicated }} clusters do not support automated [locality-aware backups](../{{site.versions["stable"]}}/take-and-restore-locality-aware-backups.html). However, user managed locality-aware backups using user provided storage locations are supported in {{ site.data.products.serverless }}, {{ site.data.products.dedicated }}, and {{ site.data.products.core }} clusters. That is, you need to configure and manage your own locality-aware backups.
> 
> ## Performance
> ```

@vy-ton can you help @ianjevans explain the consequences of this unsupported feature in terms of UX
___
*[`cockroachcloud/serverless-unsupported-features.md` line 32 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4byEAv5zlyHokgxJRW:-N4byEAv5zlyHokgxJRX:bwl6ao6) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L32)):*
> ```Markdown
> ## Multi-region clusters
> 
> {{ site.data.products.serverless }} does not support [multi-region](../{{site.versions["stable"]}}/multiregion-overview.html) features.
> ```

Should we say something first like "Serverless is only supported in a single-region and therefore does not support multi-region"?
___
*[`cockroachcloud/serverless-unsupported-features.md` line 40 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4by_v5DAwp3Cn4-GWD:-N4by_v5DAwp3Cn4-GWE:b-9awneh) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L40)):*
> ```Markdown
> ## CockroachDB Cloud Console features
> 
> The [DB Console](../{{site.versions["stable"]}}/ui-overview.html) is not supported in {{ site.data.products.serverless }}. The CockroachDB [Cloud Console](cluster-overview-page.html) provides metrics and graphs to monitor the health, performance, and state of your cluster.
> ```

@kevin-v-ngo do you want to cover whats not in the cc console here that is in the db console in more detail?

**Review by vy-ton — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14282)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @abhinavg6, @amruss, @emily-horing, @ianjevans, @mwang1026, and @thtruo)

___
*[`cockroachcloud/serverless-unsupported-features.md` line 20 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4c4xoZE8_s1DiH32UD:-N4c4xoZE8_s1DiH32UE:b6mjuk3) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L20)):*
> ```Markdown
> ## Backups
> 
> {{ site.data.products.serverless }} only support full backups. [Incremental](../{{site.versions["stable"]}}/take-full-and-incremental-backups.html) and [revision history](../{{site.versions["stable"]}}/take-backups-with-revision-history-and-restore-from-a-point-in-time.html) backups are not supported.
> ```

Is this for Serverless as it is today or GA?

@livlobo Can you confirm this?
___
*[`cockroachcloud/serverless-unsupported-features.md` line 26 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4by5RbAY64zacQBva7:-N4c55i-27ldQzI6zsk0:b-zhbdc) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L26)):*
<details><summary><i>Previously, awoods187 (Andy Woods) wrote…</i></summary><blockquote>

@vy-ton can you help @ianjevans explain the consequences of this unsupported feature in terms of UX
</blockquote></details>

Without locality aware backups, users could have higher cost/worst performance since node data might be backed up to a different region. In addition, you might need to ensure that all data (cluster data and backup) meet data domiciling requirements. Since the key highlight is that the feature is not automated, users can solve these problems but with more effort/design/code.

**Review by abhinavg6 — APPROVED** [review]

File: cockroachcloud/serverless-unsupported-features.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+{{ site.data.products.serverless }} does not support [multi-region](../{{site.versions["stable"]}}/multiregion-overview.html) features.
+
+## Self service upgrades
+
+{{ site.data.products.serverless }} is a fully managed multi-tenant deployment of CockroachDB. Major and minor upgrades of CockroachDB are handled by Cockroach Labs, and [can't be initiated by users](serverless-faqs.html#can-i-upgrade-the-version-of-cockroachdb-my-cockroachdb-serverless-beta-cluster-is-running-on).
+
+## CockroachDB Cloud Console features
+
+The [DB Console](../{{site.versions["stable"]}}/ui-overview.html) is not supported in {{ site.data.products.serverless }}. The CockroachDB [Cloud Console](cluster-overview-page.html) provides metrics and graphs to monitor the health, performance, and state of your cluster.
```
[the remark this answers is not in the document]
Comment (kevin-v-ngo) [reply cockroachcloud/serverless-unsupported-features.md:40]: Note there is information regarding Backups in CC (which is in the Jobs DB console page)

**Review by emily-horing — APPROVED** [review]

**Review by thtruo — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14282)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @amruss, @ianjevans, @mwang1026, and @vy-ton)

___
*[`cockroachcloud/serverless-unsupported-features.md` line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4dFWN8AMXYtQluBv9-:-N4dFWN8AMXYtQluBv90:b-a6w7s0) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L38)):*
> ```Markdown
> {{ site.data.products.serverless }} is a fully managed multi-tenant deployment of CockroachDB. Major and minor upgrades of CockroachDB are handled by Cockroach Labs, and [can't be initiated by users](serverless-faqs.html#can-i-upgrade-the-version-of-cockroachdb-my-cockroachdb-serverless-beta-cluster-is-running-on).
> 
> ## CockroachDB Cloud Console features
> ```

Nit: could we rephrase this along the lines of "Monitoring" or "Monitoring workloads and cluster health" to line up better with the rest of the header languages around feature category?  



_Code quote:_
```Markdown
## CockroachDB Cloud Console features
```

**Comment by ianjevans** [comment]
@awoods187:
> Philosophical question. Do we want to use feature differences here? I'm thinking maybe something like "user experience differences" because we deliver this same UX via different features. I don't think its important to mention a feature not in serverless, but in crdb, when the UX the feature provides is provided by another feature, if it is, we should at least separate that out from other unsupported UX differences.

I agree that in some cases we have equivalent UX between Serverless and CRDB core, and we should be clear in this doc when an apparent feature gap is covered by additional Serverless features unavailable in CRDB core. Maybe we should have high-level sections on UX difference and feature gaps.

I think the intent of this doc is to set expectations for users about what is and is not available when they are looking at our distributions. E.g. someone who is interested in CRDB for its multiregion capabilities needs to understand that Serverless does not currently support our MR semantics.

> @vy-ton can you help @ianjevans explain the consequences of this unsupported feature in terms of UX

@vy-ton I think Andy wanted more details on how the lack of DistSQL impacts Serverless performance.

> Should we say something first like "Serverless is only supported in a single-region and therefore does not support multi-region"?

Done.

**Comment by ianjevans** [comment]
@vy-ton 
> Is this for Serverless as it is today or GA?

As it is today. I will update it for GA.

**Comment by ianjevans** [comment]
> Incremental and Revision history backups are not supported in Serverless for the automated managed backups taken by CRL for the customer. However, customers can take incremental and revision history backups to their own storage location manually.

Done.

File: cockroachcloud/serverless-unsupported-features.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ianjevans) [reply cockroachcloud/serverless-unsupported-features.md:40]: > The CockroachDB Cloud Console provides a subset of observability information from the DB console including SQL Metrics, SQL Activity, and Databases information. The Cloud Console does not include information from the following DB console pages:
> 
> * Non-SQL metrics
> * Network Latency
> * Hot ranges
> * Jobs
> * Advanced Debug

Done.

**Comment by ianjevans** [comment]
> > Reviewed 1 of 2 files at r1, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14282)_ status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @amruss, @ianjevans, @mwang1026, and @vy-ton)
> 
> _[`cockroachcloud/serverless-unsupported-features.md` line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/14282#-N4dFWN8AMXYtQluBv9-:-N4dFWN8AMXYtQluBv90:b-a6w7s0) ([raw file](https://github.com/cockroachdb/docs/blob/9ac786f2ceb242a422fe5cbebfd166a234821da6/cockroachcloud/serverless-unsupported-features.md#L38)):_
> 
> > ```gfm
> > {{ site.data.products.serverless }} is a fully managed multi-tenant deployment of CockroachDB. Major and minor upgrades of CockroachDB are handled by Cockroach Labs, and [can't be initiated by users](serverless-faqs.html#can-i-upgrade-the-version-of-cockroachdb-my-cockroachdb-serverless-beta-cluster-is-running-on).
> > 
> > ## CockroachDB Cloud Console features
> > ```
> 
> Nit: could we rephrase this along the lines of "Monitoring" or "Monitoring workloads and cluster health" to line up better with the rest of the header languages around feature category?
> 
> _Code quote:_
> 
> ```gfm
> ## CockroachDB Cloud Console features
> ```

Done.

**Comment by vy-ton** [comment]
> @vy-ton I think Andy wanted more details on how the lack of DistSQL impacts Serverless performance.

Sorry I misaligned the comment. Users do not benefit from the improved query performance that DistSQL offers, especially for OLAP, analytical queries.

**Comment by vy-ton** [comment]
@ianjevans Can we add a section for follower reads to this page, a [recent discovery](https://cockroachlabs.slack.com/archives/C04U1BTF8/p1655749422578809)?

**Comment by ianjevans** [comment]
> Can we add a section for follower reads to this page, a [recent discovery](https://cockroachlabs.slack.com/archives/C04U1BTF8/p1655749422578809)?

Yup, I'll clarify this in the multi-region info.

**Comment by ianjevans** [comment]
If nobody objects, I intend to merge this PR by COB today.

**Pull request merged**